app/models/alert_date.py

```diff
@@ -8,6 +14,18 @@ def __init__(self, _datetime):
         self._datetime = _datetime  # a timezone-aware UTC datetime
         self._local_datetime = utc_string_to_aware_gmt_datetime(self._datetime)
 
+    def __eq__(self, other):
+        return self._datetime == other._datetime
+
+    def __lt__(self, other):
```

Comment:
is there a reason for these function names to be abbreviated? I'm asking as it makes them less readable.

Also I can't seem to see where we use them 😳

Reply:
`__eq__` and `__lt__` are examples of ‘dunder’ (**d**ouble **under**score) methods which have a special meaning in Python. 

`__eq__` is called on an object when you use the equality operator, for example these two lines are the same:
```
AlertDate(datetime.utcnow()) == 1
AlertDate(datetime.utcnow()).__eq__(1)
```

Without defining `__eq__`:
```
now = datetime.utcnow()
AlertDate(now) == AlertDate(now)
> False
```

With `__eq__` defined:
```
now = datetime.utcnow()
AlertDate(now) == AlertDate(now)
> True
```

***

`__lt__` is called on an object when you use the less than operator, for example these two lines are the same:
```
AlertDate(datetime.utcnow()) < 1
AlertDate(datetime.utcnow()).__lt__(1)
```

***

Implementing both `__eq__` and `__lt__` makes an object _sortable_. Which means you can do this:
```
sorted([
    AlertDate(datetime.utcnow() + timedelta(hours=2)),
    AlertDate(datetime.utcnow() + timedelta(hours=1)),
    AlertDate(datetime.utcnow() + timedelta(hours=7)),
])
> [
    AlertDate(2021-11-03 11:58:41.288081),
    AlertDate(2021-11-03 12:58:41.288030),
    AlertDate(2021-11-03 17:58:41.288099),
]
```

Without `__eq__` and `__lt__` implemented you get this:
```
sorted([
    AlertDate(datetime.utcnow() + timedelta(hours=2)),
    AlertDate(datetime.utcnow() + timedelta(hours=1)),
    AlertDate(datetime.utcnow() + timedelta(hours=7)),
])
> TypeError: '<' not supported between instances of 'AlertDate' and 'AlertDate'
```

So we don’t call these methods directly, but the [algorithm of the `sorted` method](https://en.wikipedia.org/wiki/Timsort) does.

Reply:
Ahh, makes sense, thanks so much for the explanation!